src/LOAuth.ts

```diff
@@ -185,23 +191,39 @@ class LOAuth {
             redirect_uri: this.clientOptions.redirectUri
           })
         });
-        const responseJson = await response.json();
-        this.token = this.client.createToken(
-          Object.assign(
-            {
-              refresh_token: this.token.refreshToken
-            },
-            responseJson
-          )
-        );
-        this._storeTokenData(this.token);
+
+        // Only create/store tokens from a valid response
+        if (response.status === 200) {
```

Comment:
Would `response.ok` be better here? It covers all 200's. May not matter though since I assume this will always be 200.

Reply:
Seems good - I updated the PR to use that instead of the status code check